task/task.go

```diff
@@ -1,15 +1,122 @@
 package task
 
 import (
+	"path/filepath"
+
+	pkgerrors "github.com/pkg/errors"
 	"github.com/symflower/eval-dev-quality/evaluate/metrics"
 	"github.com/symflower/eval-dev-quality/evaluate/repository"
+	"github.com/symflower/eval-dev-quality/language"
+	"github.com/symflower/eval-dev-quality/log"
+	evalmodel "github.com/symflower/eval-dev-quality/model"
 	taskcontext "github.com/symflower/eval-dev-quality/task/context"
 )
 
+// Task defines a task to evaluate a repository.
 type Task interface {
 	// Identifier returns the task identifier.
 	Identifier() taskcontext.Identifier
 
 	// Run runs a task in a given repository.
 	Run(repository *repository.Repository) (assessments metrics.Assessments, problems []error, err error)
 }
+
+// GetTaskByTaskIdentifier returns a task based on the task identifier.
+func GetTaskByTaskIdentifier(taskIdentifier taskcontext.Identifier, logger *log.Logger, resultPath string, model evalmodel.Model, language language.Language) (task Task, err error) {
+	switch taskIdentifier {
+	case taskcontext.IdentifierWriteTests:
+		return NewTaskWriteTests(logger, resultPath, model, language), nil
+	default:
+		return nil, taskcontext.ErrTaskUnsupported
+	}
+}
+
+// TaskWriteTests holds the write test task.
+type TaskWriteTests struct {
```

Comment:
The task interface and its implementations should be separated IMO. Ideally `Context`, `Identifier` and `Task` itself should be in `github.com/symflower/eval-dev-quality/task`. And the implementations could be e.g. in `github.com/symflower/eval-dev-quality/evaluate/task`.

Reply:
Does this resolve some of the cyclic imports you experienced?

Reply:
@bauersimon Thoughts?

Reply:
>  Ideally Context, Identifier and Task itself should be in github.com/symflower/eval-dev-quality/task

I suppose only the type `Identifier` should be in the `task` package. `IdentifierWriteTests` should then probably be in `evaluate/task`.

Reply:
The cyclic imports should be gone as long as the `Identifier` and `Context` are in a separate package, cause both the task logic and the model need them, so they cannot be in the same package than the task logic.

Hmmmm. Splitting interface and implementation to completely different packages seems a bit weird. And honestly I think a `task` is such an essential part of `evaluate` that I would move everything there. And then we would have `evaluate/task` and `evaluate/task/context`.